Extension/src/common.ts

```diff
@@ -304,7 +304,7 @@ export function resolveCachePath(input: string | undefined, additionalEnvironmen
     return resolvedPath;
 }
 
-export function resolveVariables(input: string | undefined, additionalEnvironment?: { [key: string]: string | string[] }): string {
+export function resolveVariables(input: string | undefined, additionalEnvironment?: { [key: string]: string | string[] }, arrayResults?: string[]): string {
```

Comment:
Another option is to return `{result, arrayResults}` if you want to continue to do an either-or. You'll just need to update the callers.

Reply:
I think this is the best approach. I considered returning a string[]. There's absolutely nothing wrong with adding to the arrayResults object. It's the same as passing a vector by non-const reference in C++ and having the method insert stuff. I didn't want to have to change any of the (many) existing callers of resolveVariables that expect only a string to be returned.

Reply:
If you keep it this way, please document the function and the side effects that happen to the arg.  It is my opinion that not wanting to update callers is not a good reason to do it this way.

Reply:
I added a comment. Only 1 out of the 24 callers of resolveVariables uses the arrayResults, so I didn't want to potentially risk breaking the other 23 callers and/or have to mess with changing all the usages.

Reply:
The comment is incorrect. A `string[]` result has always been possible, but would be returned as a delimited string instead.  The comment should say that if the caller needs array results returned as an array instead of a delimited string, use the `arrayResults` parameter to provide a location for the non-joined array results.

And on that note, `newValue` could be set the same in both cases. It's just an alternate form of the result.

Reply:
No, the previous result was string and not string[]. A delimited string is not a string[].

I intentionally set newValue to "" if it's not used.

Reply:
I think you misunderstood my comment. The `string[]` result I'm referring to is packed as a `;` delimited string to match the return type and unpacked later. It's still a `string[]` at heart.

Reply:
What I'm trying to say is that the comment should indicate that when `arrayResults` is passed, the result should be returned as its intended type (a `string[]`) instead of a `;` or `:` delimeted `string` and that result is copied into `arrayResults`.

Reply:
I'm expecting the reader to interpret "string[]" as a reference to the literal type and not a reference to a "string" that is a "string[] at heart". There is no need to refer to any details about the contents of the strings.

Reply:
The way I read the comment, it was confusing. Can you just change it to this and I'll resolve the comment?

`Pass in 'arrayResults' if a string[] result is possible and a delimited string result is undesirable. The string[] result will be copied into 'arrayResults'.`

Reply:
Sure